tools/data_integrity/update_candidate_latest_diagnosis.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * This script updates each candidate's latest diagnosis based on the
+ * confidured diagnosis evolution trajectory.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   Various <example@example.com>
+ * @license  Loris license
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+require_once __DIR__ . '/../generic_includes.php';
+
+use LORIS\StudyEntities\Candidate\CandID;
+
+$candIDs = $DB->pselectCol(
+    "SELECT CandID FROM candidate
+    WHERE Entity_type='Human' AND Active='Y'",
+    []
+);
+
+// get configured diagnosis trajectories in DESC order by project
+$diagnosisTrajectories = $DB->pselect(
+    "SELECT * FROM diagnosis_evolution
+    ORDER BY orderNumber DESC",
+    []
+);
+
+// if no diagnosis trajectories are defined, return null
+if (is_null($diagnosisTrajectories)) {
+    echo "There are no configured Diagnosis Trajectories. Nothing to update.\n";
+    exit;
+}
+
+// Format trajectories by project
+$diagnosisTrajectoryByProject = [];
+foreach ($diagnosisTrajectories as $key => $data) {
+    $diagnosisTrajectoryByProject[$data['ProjectID']][] = $data;
+}
+
+$loris = new \LORIS\LorisInstance(
+    \NDB_Factory::singleton()->database(),
+    \NDB_Factory::singleton()->config(),
+    [
+        "project/modules",
+        "modules",
+    ]
+);
+
+foreach ($candIDs as $k => $candID) {
+    $candidate       = \Candidate::singleton(new CandID($candID));
+    $candidateVisits = $candidate->getListOfVisitLabels();
+
+    foreach ($diagnosisTrajectoryByProject as $projectID => $projectTrajectories) {
+        foreach ($projectTrajectories as $key => $data) {
+            // search if candidate has a matching visit
+            $sessionID = array_search(
+                $data['visitLabel'],
+                array_reverse($candidateVisits, true)
+            );
+            if ($sessionID) {
+                // Get projectID of this session
+                $timepoint = \TimePoint::singleton(new SessionID($sessionID));
+                $projectID = $timepoint->getProjectID();
```

Comment:
is this $projectID the same as the $projectID defined in the foreach on line 56?

Reply:
yes! I'll update